transaction-service/src/main/java/pl/adamboguszewski/transaction/service/infrastructure/controller/TransactionController.java

```diff
@@ -31,8 +31,8 @@ public TransactionController(TransactionService transactionService) {
     }
 
     @GetMapping("/{id}")
-    public ResponseEntity getTransactionById(@PathVariable Long id) {
-        return new ResponseEntity(HttpStatus.OK);
+    public ResponseEntity<Optional<Transaction>> getTransactionById(@PathVariable Long id) {
```

Comment:
We don't return optional here. If transaction is null it will return just empty json which is good: { }